MAVProxy/modules/mavproxy_link.py

```diff
@@ -541,7 +541,7 @@ def emit_accumulated_statustext(self, key, id, pending):
 
     def master_msg_handling(self, m, master):
         '''link message handling for an upstream link'''
-        if self.settings.target_system != 0 and m.get_srcSystem() != self.settings.target_system:
+        if not self.is_primary_vehicle(m):
```

Comment:
This will break multi-vehicle-capable modules, won't it?

Reply:
Tested in SITL with 10-copters and the "swarm" module. All works as expected